### PR TITLE
Amend/caching #128

### LINKED ISSUE #128: Add Spring Caching and some Cache implementation
We should prepare a caching implementation that can be used with Spring's Caching abstraction so that we can just use it whereever we please.

@outcobra/developers What implementation should we use?

### PR DESCRIPTION
**Pull request by needToRoll**
This closes #128

Summary of changes:
- ​updated kotlin and related dependencies
- added caching #128 

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] Update CI script and configs on Jenkins (`Jenkinsfile`, `auth0.yml`, `application.yml`, etc.)

Please review @jmesserli / cc: @outcobra/developers.

### CODE REVIEW

**Comment by M4R1KU** [comment]
Does it work?

**Comment by needToRoll** [comment]
No I just implemented it, committed it, pushed it and opened a pr for something that does not work!

**Comment by M4R1KU** [comment]
Hmmm it does not work

**Comment by needToRoll** [comment]
like hell it does

**Comment by M4R1KU** [comment]
Nope

File: backend/src/main/kotlin/outcobra/server/util/RepositoryLocator.kt [hunk 1]
```diff
@@ -26,10 +31,10 @@ class RepositoryLocator @Inject constructor(val context: ApplicationContext) {
      * @see [BeanFactory.getBean]
      * @since 1.0.0
      */
-    //@Cacheable("repoForName")
+    @Cacheable(REPO_FOR_NAME)
     fun getForEntityName(entityName: String): JpaRepository<*, Long> {
         val repoName = entityName.firstToLower() + "Repository"
-
+        LoggerFactory.getLogger(this::class.java).info("function called with $entityName")
```
Comment (jmesserli) [inline backend/src/main/kotlin/outcobra/server/util/RepositoryLocator.kt]: Remove this

**Pull request merged**